starlite/handlers/http.py

```diff
@@ -280,24 +281,43 @@ def _get_response_from_data(
         """
         Determines the correct response type to return given data
         """
-        if isinstance(data, Redirect):
-            return RedirectResponse(headers=headers, status_code=self.status_code, url=data.path)
-        if isinstance(data, File):
-            return FileResponse(media_type=media_type, headers=headers, **data.dict())
-        if isinstance(data, Stream):
-            return StreamingResponse(
-                content=data.iterator, status_code=self.status_code, media_type=media_type, headers=headers
-            )
-        if isinstance(data, Template):
-            if not app.template_engine:
-                raise ImproperlyConfiguredException("Template engine is not configured")
-            return TemplateResponse(
-                context=data.context,
-                template_name=data.name,
-                template_engine=app.template_engine,
-                status_code=self.status_code,
-                headers=headers,
-            )
+        if isinstance(data, (Redirect, File, Stream, Template)):
```

Comment:
Could make `StarliteType` abstract with a `to_starlette_response()` method:

```python
# datastructures.py
class StarliteType(BaseModel, abc.ABC):
    ...
    @abc.abstractmethod
    def to_starlette_response(self, media_type: MediaType, headers: dict[str, str], status_code: int) -> StarletteResponse:
        """
        Construct a `starlette.responses.Response` appropriate for type.

        Args:
            media_type (MediaType): Media type for response.
            headers (dict[str, str]): Headers to be merged with `self.headers` and added to the response.
            status_code (int): Status code for response.

        Returns:
            StarletteResponse
        """
        ...


class File(StarliteType):
    ...
    def to_starlette_response(self, media_type: MediaType, headers: dict[str, str], status_code: int) -> FileResponse:
        return FileResponse(
            background=self.background,
            filename=self.filename,
            headers={**headers, **self.headers},
            media_type=media_type,
            path=self.path,
            stat_result=self.stat_result,
        )
```
This method would then collapse down to:

```python
    def _get_response_from_data(
        self,
        headers: dict,
        data: Union[StarletteResponse, StarliteType],
        media_type: Union[MediaType, str],
        app: "Starlite",
) -> StarletteResponse:
    if isinstance(data, StarliteType):
        return data.to_starlette_response(media_type, headers, self.status_code)
    return data
```
Could even do away with this method altogether in this case and do that instance check in `to_response()`.

Reply:
maybe in a followup.